python/lsst/daf/butler/core/fileTemplates.py

```diff
@@ -106,13 +106,13 @@ class FileTemplate:
     def __init__(self, template):
         self.template = template
 
-    def format(self, dataUnits, datasetType=None, component=None):
+    def format(self, dataId, datasetType=None, component=None):
```

Comment:
I think this may not need a `component` argument anymore, since that's packed into the `datasetType`.

In the future, it will probably need to have a `run` argument, as we'll need templates to be unique across runs as well.

Reply:
Yes. Now that I've encoded the dot syntax into `DatasetType` I don't think a separate component is needed any more. I think I need to use the new `DatasetType` splitter here now.

Reply:
Actually, I think it should just take a `DatasetRef` directly (it has all the information).